R/LearnerClassifXgboost.R

```diff
@@ -314,28 +316,41 @@ LearnerClassifXgboost = R6Class("LearnerClassifXgboost",
         measure = pv$eval_metric
 
         fun = if (pv$objective == "binary:logistic" && measure$predict_type == "prob" && inherits(measure, "MeasureBinarySimple")) {
-            xgboost_binary_binary_prob
-          } else if (pv$objective == "binary:logistic" && measure$predict_type == "prob" && inherits(measure, "MeasureClassifSimple")) {
-            xgboost_binary_classif_prob
-          } else if (pv$objective == "binary:logistic" && measure$predict_type == "response") {
-            xgboost_binary_response
-          } else if (pv$objective == "multi:softprob" && measure$predict_type == "prob") {
-            xgboost_multiclass_prob
-          } else if (pv$objective %in% c("multi:softmax", "multi:softprob") && measure$predict_type == "response") {
-            xgboost_multiclass_response
-          } else {
-            stop("Only 'binary:logistic', 'multi:softprob' and 'multi:softmax' objectives are supported.")
-          }
+          xgboost_binary_binary_prob
+        } else if (pv$objective == "binary:logistic" && measure$predict_type == "prob" && inherits(measure, "MeasureClassifSimple")) {
+          xgboost_binary_classif_prob
+        } else if (pv$objective == "binary:logistic" && measure$predict_type == "response") {
+          xgboost_binary_response
+        } else if (pv$objective == "multi:softprob" && measure$predict_type == "prob") {
+          xgboost_multiclass_prob
+        } else if (pv$objective %in% c("multi:softmax", "multi:softprob") && measure$predict_type == "response") {
+          xgboost_multiclass_response
+        } else {
+          stop("Only 'binary:logistic', 'multi:softprob' and 'multi:softmax' objectives are supported.")
+        }
 
-        pv$eval_metric =  mlr3misc::crate({function(pred, dtrain) {
-            scores = fun(pred, dtrain, measure, n_classes)
-             list(metric = measure$id, value = scores)
-          }}, n_classes, measure, fun)
+        pv$custom_metric =  mlr3misc::crate({function(pred, dtrain) {
+          scores = fun(pred, dtrain, measure, n_classes)
+          list(metric = measure$id, value = scores)
+        }}, n_classes, measure, fun)
 
         pv$maximize = !measure$minimize
       }
 
-      invoke(xgboost::xgb.train, data = xgb_data, .args = pv)
+      xgboost::xgb.train(
```

Comment:
Are we sure there is no other `pv` values we might need as `.args`?

Reply:
`xgboost::xgb.params` contains all allowed parameters

Reply:
This function is new in xgboost. https://xgboost.readthedocs.io/en/latest/R-package/migration_guide.html